src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,91 @@
+package duke;
+
+import duke.taskings.Task;
+import duke.taskings.TaskList;
+import duke.messages.Ui;
+
+import java.util.ArrayList;
+
+public class Parser {
+    static final boolean EXIT = false;
+    static final boolean CONTINUE = true;
+
+    /**
+     * Returns the number of words within a string.
+     *
+     * @param input the string input from the user
+     * @return the number of words in the string
+     */
+    public static int getNumOfWords(String input) {
+        if (input == null || input.isEmpty()) {
+            return 0;
+        }
+        String[] words = input.split("\\s+");
+        return words.length;
+    }
+
+    /**
+     * parses the user input and processes it with the tasks array list.
+     *
+     * @param inData the user input .
+     * @param tasks the array which would be operated on.
+     *
+     * @return EXIT if input equals "bye", else return CONTINUE
+     */
+
+    public static boolean processInput(String inData, ArrayList<Task> tasks) {
+
+        if (inData.equals("help")) {
+            Ui.getHelp();
+        } else if (inData.equals("list")) {
+            //print entire list if input is equal to "list"
+            TaskList.getList(tasks);
+        } else if (inData.equals(("bye"))) {
+            Ui.showExit();
+            return EXIT; //exit loop
+        } else if (inData.isBlank() || inData.isEmpty()) {
+            Ui.showWrongCommand();
+        } else if (inData.contains(" ")) {
+            //further parse the user input for user taskings
+            int numOfWords = getNumOfWords(inData);
+            String command = inData.split(" ")[0];
+            String[] userInput = inData.split(" ");
+            StringBuilder synthesizedArr = new StringBuilder();
+            for (int i = 1; i < numOfWords; i += 1) {
+                synthesizedArr.append(userInput[i]).append(" ");
+            }
```

Comment:
`getNumOfWords` splits `words` on arbitrary whitespace, but here you split only on " ". There _might_ be a bug here. It might make more sense to return `words` from the method and build the `synthesizedArr` directly from that.

Then you would only need to pass `words` to `processCommand` rather than both `numOfWords` and `userInput`.